pyretailscience/segmentation/segstats.py

```diff
@@ -253,26 +253,32 @@ def _calc_seg_stats(
         if calc_rollup:
             rollup_metrics = []
 
-            # Configuration for both prefix and suffix rollups
+            # Configuration for rollups
             rollup_configs = [
-                # Prefix rollups: group by prefixes, mutate remaining columns
-                [(segment_col[:i], segment_col[i:], rollup_value[i:]) for i in range(1, len(segment_col))],
-                # Suffix rollups: group by suffixes, mutate preceding columns
-                [(segment_col[i:], segment_col[:i], rollup_value[:i]) for i in range(1, len(segment_col))],
+                # Prefix rollups: always include when calc_rollup=True
+                *[(segment_col[:i], segment_col[i:], rollup_value[i:]) for i in range(1, len(segment_col))],
             ]
 
+            # Only add suffix rollups when calc_total=True (to avoid "Total" in category when no grand total)
+            if calc_total:
+                rollup_configs.extend(
+                    [
+                        # Suffix rollups: group by suffixes, mutate preceding columns
+                        *[(segment_col[i:], segment_col[:i], rollup_value[:i]) for i in range(1, len(segment_col))],
+                    ],
```

Comment:
This can be simplified to,
```suggestion
                rollup_configs.extend(
                    # Suffix rollups: group by suffixes, mutate preceding columns                
                    [
                       (segment_col[i:], segment_col[:i], rollup_value[:i]) for i in range(1, len(segment_col))
                    ],
```